connect/src/main/java/com/stripe/android/connect/StripeComponentDialogFragment.kt

```diff
@@ -19,6 +19,68 @@ import kotlinx.coroutines.flow.filterNotNull
 import kotlinx.coroutines.flow.first
 import kotlinx.coroutines.launch
 
+/**
+ * A full-screen DialogFragment that displays a full-screen component.
+ *
+ * The implementation to make this all work *well* is quite tricky and deserves some explanation.
```

Comment:
👀 These implementation notes are from `StripeComponentControllerImpl`. I didn't put this in `StripeComponentController` due to previous feedback to reduce its visibility to users.